### ISSUE TITLE
Running gulp from package.json generates flags.json in root

### ISSUE THREAD

**Issue by charliegeiger89** [gulpjs/gulp#934]
*Gulp version 3.8.11*

Running Gulp from package.json, 'npm run gulp', results in the generation of **3.28.73.flags.json** in the root directory. CD to the gulpfile.js directory and running 'gulp' does not cause the flags.json to be generated.

**Directory tree:**
```
ExampleApp/
├── 3.28.73.flags.json
├── node_modules
├── source
├── build
│   └── gulp
│       ├── config.js
│       ├── gulpfile.js
│       └── tasks
│           ├── deployment
│           ├── development
│           │   └── default.js
│           └── testing
└── package.json
```
**package.json scripts:**
```
  "scripts": {
    "gulp": "gulp --cwd build/gulp/"
  }
```

**Comment by heikki** [comment]
Can you try [this repo](https://github.com/heikki/npm-run-gulp-debug) and run `npm run gulp` to see if the same thing happens?

**Comment by charliegeiger89** [comment]
Just cloned and the same issue occurred. I ran ```npm run gulp``` without running ``` npm i``` ( no local gulp ) and it still output the flags.json file. I re-cloned, ran ```npm i```, then ```npm run gulp``` and it also still output the flags.json file.

**Comment by heikki** [comment]
- What os, what versions of node & npm?
- How node is installed? If with nvm or similar: which & what version?
- Do you get any errors when running `npm run gulp` without local install?

**Comment by heikki** [comment]
My output without local install:

```sh
∴ npm-run-gulp-debug git:(master) npm run gulp

> @ gulp /Users/heikki/Desktop/npm-run-gulp-debug
> gulp --cwd build/gulp/

[13:43:02] Working directory changed to ~/Desktop/npm-run-gulp-debug/build/gulp

Error: Cannot find module 'orchestrator'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/heikki/node_modules/gulp/index.js:4:20)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/Users/heikki/.nvm/v0.10.36/bin/npm" "run" "gulp"
npm ERR! node v0.10.36
npm ERR! npm  v2.6.1
npm ERR! code ELIFECYCLE
npm ERR! @ gulp: `gulp --cwd build/gulp/`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the @ gulp script 'gulp --cwd build/gulp/'.
npm ERR! This is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp --cwd build/gulp/
npm ERR! You can get their info via:
npm ERR!     npm owner ls 
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/heikki/Desktop/npm-run-gulp-debug/npm-debug.log
```

**Comment by charliegeiger89** [comment]
* CentOS 6.6
* Node 0.12.0
* NPM 2.5.1

Node was installed by compiling from source.

Error output without local install:
```
> @ gulp /var/www/html/npm-run-gulp-debug
> gulp --cwd build/gulp/

[17:58:26] Local gulp not found in /var/www/html/npm-run-gulp-debug/build/gulp
[17:58:26] Try running: npm install gulp

npm ERR! Linux 2.6.32-504.el6.x86_64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "gulp"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE
npm ERR! @ gulp: `gulp --cwd build/gulp/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ gulp script 'gulp --cwd build/gulp/'.
npm ERR! This is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp --cwd build/gulp/
npm ERR! You can get their info via:
npm ERR!     npm owner ls
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /var/www/html/npm-run-gulp-debug/npm-debug.log
```

**Comment by charliegeiger89** [comment]
Updated NPM to match your version of **2.6.1**.

Error output without local install:
```
> @ gulp /var/www/html/npm-run-gulp-debug
> gulp --cwd build/gulp/

[18:08:57] Local gulp not found in /var/www/html/npm-run-gulp-debug/build/gulp
[18:08:57] Try running: npm install gulp

npm ERR! Linux 2.6.32-504.el6.x86_64
npm ERR! argv "node" "/usr/local/bin/npm" "run" "gulp"
npm ERR! node v0.12.0
npm ERR! npm  v2.6.1
npm ERR! code ELIFECYCLE
npm ERR! @ gulp: `gulp --cwd build/gulp/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ gulp script 'gulp --cwd build/gulp/'.
npm ERR! This is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp --cwd build/gulp/
npm ERR! You can get their info via:
npm ERR!     npm owner ls
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /var/www/html/npm-run-gulp-debug/npm-debug.log
```

**Comment by heikki** [comment]
Ping @tkellen 

This might be interesting. I have no clue what could cause that :smile:

**Comment by tkellen** [comment]
@charliegeiger89 can you gist the output of `require('os').tmpdir()` on your installation?

@heikki, this is causing it:
https://github.com/tkellen/js-v8flags/blob/master/index.js#L6

also, here is the code for `os.tmpdir`
```js
exports.tmpdir = function() {
  if (isWindows) {
    return process.env.TEMP ||
           process.env.TMP ||
           (process.env.SystemRoot || process.env.windir) + '\\temp';
  } else {
    return process.env.TMPDIR ||
           process.env.TMP ||
           process.env.TEMP ||
           '/tmp';
  }
};
```

**Comment by charliegeiger89** [comment]
Are you looking for this?

```
> console.log( require('os').tmpdir() );
/tmp
undefined
```

**Comment by tkellen** [comment]
@charliegeiger89 Hrm. Yes. Can you try running this from the gulp task?
https://github.com/heikki/npm-run-gulp-debug/pull/1/files

**Comment by charliegeiger89** [comment]
@tkellen Here is the gist: https://gist.github.com/charliegeiger89/40d8a2c4d7114c51f7df
I posted outputs for running gulp from the gulpfile.js directory and running gulp from npm run gulp.

**Comment by dimsmol** [comment]
Having the same issue with exactly same file name.
In my case babel.json file is also generated at the same time as 3.28.73.flags.json - not sure how are they related.

babel.json contains something like:

```
{
  "/some_path/gulp_tasks_lib.js:{\"experimental\":true}": {
    "code": "\"use strict\";\n\nvar _interopRequire = function (obj) { return obj && obj.__esModule ? obj[\"default\"] : obj; };\n\nvar _prototypeProperties = funct...very long string here...",
    "map": {
      "version": 3,
      "sources": [
        "/some_path/gulp_tasks_lib.js"
      ],
      "names": [],
      "mappings": ";;;;;;;;IAAO,KAAK,2BAAM,YAAY;;IACvB,GAAG,2BAAM,K...very long string here...",
      "file": "/some_path/gulp_tasks_lib.js",
      "sourcesContent": [
        "import babel from 'gulp-babel'\nimport del from 'del'\nimport runSequence from 'run-sequence...very long string here..."
      ]
    },
    "ast": null,
    "mtime": 1425338760000
  }
}
```

**Comment by phated** [comment]
@tkellen also opened an issue for this over on the v8flags repo https://github.com/tkellen/js-v8flags/issues/13

**Comment by rubennorte** [comment]
This issue may be related to this one: https://github.com/npm/npm/issues/4531

**Comment by phated** [comment]
This should be fixed in 2.0.3 according to https://github.com/tkellen/js-v8flags/issues/13#issuecomment-88298650
